python3Packages.pydantic-core: init at 2.1.3

blaggacao opened the pull request:
###### Description of changes


This package provides the core functionality for pydantic V2 validation and serialization.
Pydantic-core is currently around 17x faster than pydantic V1.

When this package is reviewed, attested and merged, we can start considering to bump pydantic to V2.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

dotlambda committed:
python310Packages.yarl: 1.8.2 -> 1.9.2

Hopefully fixes segfaults on aarch64,
https://github.com/aio-libs/yarl/issues/835

Also apply upstream patch to aiohttp to make it work with this
version of yarl.

Changelog: https://github.com/aio-libs/yarl/blob/v1.9.2/CHANGES.rst

Co-Authored-By: Robert Schütz <nix@dotlambda.de>

r-ryantm committed:
isocodes: 4.11.0 -> 4.15.0

r-ryantm committed:
libdc1394: 2.2.6 -> 2.2.7

r-ryantm committed:
fluidsynth: 2.3.2 -> 2.3.3

r-ryantm committed:
libassuan: 2.5.5 -> 2.5.6

SuperSandro2000 committed:
python310Packages.constantly: run tests

mweinelt committed:
curl: 8.1.1 -> 8.1.2

https://curl.se/changes.html#8_1_2

SuperSandro2000 committed:
python310Packages.numpy: run tests correct so that the build can fail

SuperSandro2000 committed:
python310Packages.pytest-timeout: remove dependency on pytest-cov

dotlambda committed:
python310Packages.pluggy: 1.0.0 -> 1.2.0

Diff: https://github.com/pytest-dev/pluggy/compare/refs/tags/1.0.0...1.2.0

Changelog: https://github.com/pytest-dev/pluggy/blob/refs/tags/1.2.0/CHANGELOG.rst

dotlambda committed:
python310Packages.pluggy: run tests

Commit:
Revert "gcc: kludge to prevent mass-rebuild"

This reverts commit 8dce1f7383b064ae5adee7702a8eeed06346dd24.

tobim committed:
boost: 1.79 -> 1.81

NickCao committed:
libde265: 1.0.11 -> 1.0.12

Diff: https://github.com/strukturag/libde265/compare/v1.0.11...v1.0.12

r-ryantm committed:
p11-kit: 0.24.1 -> 0.25.0

PedroHLC committed:
nix-prefetch-git: add sri-hash to output

SuperSandro2000 committed:
pypy3Packages.protobuf: fix build

tjni committed:
python310Packages.requests-mock: 1.10.0 -> 1.11.0

bobby285271 committed:
vala: Merge graphviz patches into one

I don't see a reason to separate the gvc-compact patch, we are actually already
maintaining our graphviz patch for a while (since openembedded-core is now simply
disabling valadoc).

bobby285271 committed:
vala: 0.56.7 → 0.56.9

https://gitlab.gnome.org/GNOME/vala/-/compare/0.56.7...0.56.9

r-ryantm committed:
openresolv: 3.12.0 -> 3.13.2

mweinelt committed:
python310Packages.exceptiongroup: 1.1.0 -> 1.1.2

https://github.com/agronholm/exceptiongroup/blob/1.1.2/CHANGES.rst

SuperSandro2000 committed:
python310Packages.Cython: 0.29.34 -> 0.29.36

https://github.com/cython/cython/blob/0.29.36/CHANGES.rst

jtojnar committed:
libxml2: 2.10.4 → 2.11.4

https://gitlab.gnome.org/GNOME/libxml2/-/compare/2.10.4...2.11.4

jtojnar committed:
gjs: 1.76.0 → 1.76.2

https://gitlab.gnome.org/GNOME/gjs/-/compare/1.76.0...1.76.2

jtojnar committed:
glib-networking: 2.76.0 → 2.76.1

https://gitlab.gnome.org/GNOME/glib-networking/-/compare/2.76.0...2.76.1

jtojnar committed:
gupnp_1_6: 1.6.3 → 1.6.4

https://gitlab.gnome.org/GNOME/gupnp/-/compare/gupnp-1.6.3...gupnp-1.6.4

jtojnar committed:
libxslt: 1.1.37 → 1.1.38

https://gitlab.gnome.org/GNOME/libxslt/-/compare/v1.1.37...v1.1.38

jtojnar committed:
networkmanager: 1.42.6 → 1.42.8

https://gitlab.freedesktop.org/NetworkManager/NetworkManager/-/compare/1.42.6...1.42.8

jtojnar committed:
tracker: 3.5.1 → 3.5.3

https://gitlab.gnome.org/GNOME/tracker/-/compare/3.5.1...3.5.3

jtojnar committed:
itstool: 2.0.6 → 2.0.7

https://github.com/itstool/itstool/compare/2.0.6...2.0.7

jtojnar committed:
mutest: unstable-2019-08-26 → 0-unstable-2023-02-24

https://github.com/ebassi/mutest/compare/e6246c9ae4f36ffe8c021f0a80438f6c7a6efa3a...18a20071773f7c4b75e82a931ef9b916b273b3e5

- Use new version schema.
- Add update script.
- Use more convenient homepage.

jtojnar committed:
upower: 1.90.0 → 1.90.2

- Fixes flaky test with libgupower 238.
- Adds support for installed tests.

https://gitlab.freedesktop.org/upower/upower/-/compare/v1.90.0...v1.90.1
https://gitlab.freedesktop.org/upower/upower/-/compare/v1.90.1...v1.90.2

jtojnar committed:
librsvg: 2.55.1 → 2.56.2

https://gitlab.gnome.org/GNOME/librsvg/-/compare/2.55.1...2.56.2

jtojnar committed:
libqmi: 1.32.2 → 1.32.4

Diff: https://gitlab.freedesktop.org/mobile-broadband/libqmi/-/compare/1.32.2...1.32.4

Changelog: https://gitlab.freedesktop.org/mobile-broadband/libqmi/-/blob/1.32.4/NEWS

jtojnar committed:
stoken: Clean up

- Format the expression.
- Use static repo name.
- Use autoreconfHook.

jtojnar committed:
stoken: 0.92 → 0.93

https://github.com/cernekee/stoken/compare/v0.92...v0.93

wegank committed:
openldap: drop unused db

mweinelt committed:
libxcrypt: 4.4.35 -> 4.4.36

https://github.com/besser82/libxcrypt/blob/v4.4.36/NEWS

mweinelt committed:
python310Packages.sqlalchemy: 2.0.17 -> 2.0.18

https://github.com/sqlalchemy/sqlalchemy/releases/tag/rel_2_0_18

wegank committed:
Merge pull request #241755 from wegank/openldap-mdb

openldap: drop unused db

mweinelt committed:
Merge pull request #241779 from mweinelt/libxcrypt-4.4.36

libxcrypt: 4.4.35 -> 4.4.36

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
umockdev: Move libgudev tests into passthru

libgudev is needed for an optional test but it itself relies on umockdev for testing.

mweinelt committed:
python310Packages.pillow: 9.4.0 -> 9.5.0

https://github.com/python-pillow/Pillow/blob/9.5.0/CHANGES.rst

mweinelt committed:
Merge pull request #241782 from mweinelt/sqlalchemy-2.0.18

python310Packages.sqlalchemy: 2.0.17 -> 2.0.18

mweinelt committed:
Merge pull request #241700 from jtojnar/libxml2

libxml2: 2.10.4 → 2.11.4

mweinelt committed:
Merge pull request #241701 from jtojnar/itstool

itstool: 2.0.6 → 2.0.7

mweinelt committed:
Merge pull request #241490 from mweinelt/exceptiongroup-1.1.2

python310Packages.exceptiongroup: 1.1.0 -> 1.1.2

mweinelt committed:
Merge pull request #241270 from mweinelt/django-3.2.20

python310Packages.django_3: 3.2.19 -> 3.2.20

mweinelt committed:
Merge pull request #240840 from r-ryantm/auto-update/cups

cups: 2.4.5 -> 2.4.6

wegank committed:
Merge pull request #239748 from r-ryantm/auto-update/fluidsynth

fluidsynth: 2.3.2 -> 2.3.3

wegank committed:
Merge pull request #239840 from r-ryantm/auto-update/libassuan

libassuan: 2.5.5 -> 2.5.6

wegank committed:
Merge pull request #241419 from r-ryantm/auto-update/openresolv

openresolv: 3.12.0 -> 3.13.2

mweinelt committed:
Merge pull request #241571 from SuperSandro2000/cython

python310Packages.Cython: 0.29.34 -> 0.29.36

wegank committed:
Merge pull request #240925 from r-ryantm/auto-update/p11-kit

p11-kit: 0.24.1 -> 0.25.0

jtojnar committed:
libgudev: Enable tests

jtojnar committed:
libgudev: 237 → 238

https://gitlab.gnome.org/GNOME/libgudev/-/compare/237...238

wegank committed:
Merge pull request #237472 from r-ryantm/auto-update/libdc1394

libdc1394: 2.2.6 -> 2.2.7

wegank committed:
Merge pull request #235887 from r-ryantm/auto-update/isocodes

isocodes: 4.11.0 -> 4.15.0

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #240409 from SuperSandro2000/numpy-fail-tests

python310Packages.numpy: run tests correct so that actually fail the build

trofi committed:
gcc: restore 'Configured with:' output on 'gcc -v' output

When reporting 'gcc' bugs upstream I'm occasionally asked for the actual
configure flags I used to build gcc.

Before the change `gcc -v` always reported empty string:

    $ gcc -v
    ...
    Target: x86_64-unknown-linux-gnu
    Configured with:

After the change `gcc -v` contains original options with slightly
mangled nix store paths to avoid build-only dependency retention:

    $ gcc -v
    ...
    Target: x86_64-unknown-linux-gnu
    Configured with: ../source/configure --prefix=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gcc-14.0.0 --with-gmp-include=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gmp-6.2.1-dev/include --with-gmp-lib=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-gmp-6.2.1/lib --with-mpfr-include=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-mpfr-4.2.0-dev/include --with-mpfr-lib=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-mpfr-4.2.0/lib --with-mpc=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-libmpc-1.3.1 --with-native-system-header-dir=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-glibc-2.37-8-dev/include --with-build-sysroot=/ --program-prefix= --enable-lto --disable-libstdcxx-pch --without-included-gettext --with-system-zlib --enable-checking=release --enable-static --enable-languages=c,c++ --disable-multilib --enable-plugin --disable-libcc1 --with-isl=/nix/store/eeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee-isl-0.20 --disable-bootstrap --build=x86_64-unknown-linux-gnu --host=x86_64-unknown-linux-gnu --target=x86_64-unknown-linux-gnu

While a bit verbose I think it's useful to see how we build gcc.

reckenrode committed:
db: make patches and autoreconfHook unconditional

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
librdf_raptor2: backport fix for `libxml2-2.11`

`libxml2-2.11` removed `checked` bit of private API and broke
`librdf_raptor2` build as:

       src/raptor_libxml.c:256:12: error: 'xmlEntity' {aka 'struct _xmlEntity'} has no member named 'checked'
         256 |     if(!ret->checked)
             |            ^~

The change pull upstream fix.

mweinelt committed:
Merge pull request #242012 from trofi/librdf_raptor2-libxml2-fix

librdf_raptor2: backport fix for `libxml2-2.11`

mweinelt committed:
Merge pull request #240236 from mweinelt/curl-8.1.2

curl: 8.1.1 -> 8.1.2

github-actions[bot] committed:
Merge staging-next into staging

reckenrode committed:
rustc: remove Darwin special-casing

An attempt to remove it was made in #106886, but there was a build
failure. Since that failure no longer occurs, remove it now.

Based on thefloweringash/nixpkgs@19042366485d2141daad7ec3e0dd1ec309cc78b1.

teto committed:
luarocks: can be embedded with lua.withPackages

lua.withPackages only brings packages that have passthru.luaModule, which was not the case of luarocks (because it's built differently than other lua packages). We call toLuaModule on luarocks so that it is brought in a lua.withPackages call

misuzu committed:
python310Packages.orjson: 3.8.11 -> 3.9.2

https://github.com/ijl/orjson/releases/tag/3.9.2
https://github.com/ijl/orjson/compare/3.8.11...3.9.2

Changelog: https://github.com/ijl/orjson/blob/3.9.2/CHANGELOG.md

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #242060 from reckenrode/bdb-clang16-mk2

db: make patches and autoreconfHook unconditional

mweinelt committed:
Merge pull request #241913 from mweinelt/pillow-9.5.0

python310Packages.pillow: 9.4.0 -> 9.5.0

github-actions[bot] committed:
Merge staging-next into staging

AndersonTorres committed:
pythonPackages.importlib-metadata: 6.0.0 -> 6.6.0

Triggered by the update of yapf.

AndersonTorres committed:
pythonPackages.importlib-metadata: 6.6.0 -> 6.8.0

Also, add myself as maintainer.

AndersonTorres committed:
yapf: 0.32.0 -> 0.40.1

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
glib: 2.76.3 → 2.76.4

https://gitlab.gnome.org/GNOME/glib/-/compare/2.76.3...2.76.4

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

wegank committed:
darwin.rewrite-tbd: 20201114 -> unstable-2023-03-27

mweinelt committed:
Merge pull request #242085 from misuzu/orjson-update

python310Packages.orjson: 3.8.11 -> 3.9.2

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

league committed:
buildenv: Limit exclusion of info/dir

While trying to exclude a generated `/share/info/dir` in a package
from symlinking into a profile, we also ended up excluding the Emacs
info manual for `dired-x`.  This change excludes only the dir file.

SuperSandro2000 committed:
Merge pull request #241075 from tjni/requests-mock

python310Packages.requests-mock: 1.10.0 -> 1.11.0

SuperSandro2000 committed:
Merge pull request #240413 from SuperSandro2000/pytest-timeout

python310Packages.pytest-timeout: remove dependency on pytest-cov

blaggacao committed:
python3Packages.pydantic-core: init at 2.1.3

blaggacao commented on pkgs/development/python-modules/typing-extensions/default.nix:
https://github.com/pydantic/pydantic-core/blob/main/pyproject.toml#L36

SuperSandro2000 replied:
Can we just relax this if this is causing to many rebuilds?

blaggacao replied:
I thin we can: https://github.com/pydantic/pydantic-core/pull/728

How much do we need to care about PyPy `3.{7,8}`?

blaggacao replied:
Local `nixpkgs-review` failed on pythonRelaxDepsHook`.... Need to investigate more.

SuperSandro2000 commented on pkgs/development/python-modules/pydantic-core/default.nix:
```diff
@@ -0,0 +1,46 @@
+
+buildPythonPackage rec {
+  pname = "pydantic-core";
+  version = "2.1.3";
```
The commit for this contains an extra v in the version number

blaggacao replied:
fixed, thx

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

wegank committed:
Merge pull request #242433 from wegank/rewrite-tbd-update

darwin.rewrite-tbd: 20201114 -> unstable-2023-03-27

github-actions[bot] committed:
Merge staging-next into staging

raboof committed:
Merge pull request #234880 from raboof/yarl-1.8.2-to-1.9.2

python310Packages.yarl: 1.8.2 -> 1.9.2

bobby285271 committed:
Merge pull request #241298 from bobby285271/upd/vala

vala: 0.56.7 → 0.56.9

SuperSandro2000 committed:
Merge pull request #239947 from SuperSandro2000/constantly-tests

python310Packages.constantly: run tests

SuperSandro2000 committed:
Merge pull request #241000 from SuperSandro2000/pypy3/protobuf

pypy3Packages.protobuf: fix build

OPNA2608 committed:
fluidsynth: Fix CMake config

Upstream is concatenating CMAKE_INSTALL_LIBDIR onto CMAKE_INSTALL_PREFIX to get
a directory to use for the install_name on Darwin. This is not the right way of
making CMAKE_INSTALL_LIBDIR absolute.

The config was first set up for breakage when the CMAKE_INSTALL_LIBDIR was made relative,
in order to work around the wrong concatenation. This caused the _IMPORT_PREFIX to be
computed by walking up the directory tree from the config file, instead of using
the absolute CMAKE_INSTALL_LIBDIR.

Then outputs were introduced to the derivation and the config was moved to a different
output. Now the prefix it walks up to is the wrong prefix, and the config is busted.

Fix by reverting the original workaround so we have an absolute CMAKE_INSTALL_LIBDIR,
and pulling a patch that fixes the bad concatenation.

Artturin committed:
Merge pull request #242532 from league/builder-info-bug

fixes #242394

trofi committed:
Merge pull request #242657 from OPNA2608/fix/fluidsynth_cmake_config

fluidsynth: Fix CMake config

github-actions[bot] committed:
Merge staging-next into staging

trofi committed:
Merge pull request #241980 from trofi/gcc-print-configure-flags

gcc: restore 'Configured with:' output on 'gcc -v' output

Artturin committed:
ragel: enableParallelBuilding

ragel: 47s -> 9s
ragelDev: 1m51s  45s

blaggacao commented:
I'd temporarily put this on hold for further local testing and wiring of `pytest` hooks and reopen when I've successfully completed those tests.

martinetd committed:
libbpf: 1.2.0 -> 1.2.2

1.2.1: Bug fix release:

Single bug fix (#1) that fixes regression in `perf` tool caused by libbpf
resetting its custom catch-all `SEC()` handler on explicit
`bpf_program__set_type()` call.

Given setting custom `SEC()` handlers is rarely used and pretty
esoteric feature of libbpf, most users should not be affected.

1.2.2: One more fix:

 - Fix (#2) possible double-free in USDT-related libbpf code, which
happens when libbpf runs out of space in `__bpf_usdt_specs` map due
to having too many unique USDT specs. Running out of space can be
mitigated by bumping up `BPF_USDT_MAX_SPEC_CNT` define before including
`bpf/usdt.bpf.h` header in BPF-side code.
This will prevent the double-free as a side effect (and will make it
possible to successfully attach all requested USDTs), which is a
recommended work-around for libbpf versions prior to v1.2.2.

Link: https://github.com/libbpf/libbpf/commit/e4d3827e5d0f21a6b4b535ccc755cc41aa590237 #1
Link: https://github.com/libbpf/libbpf/commit/f117080307163d7057034341aa8ff6b201041599 #2

Artturin committed:
Merge pull request #242699 from Artturin/parallel1

jtojnar committed:
gupnp: Fix build against libxml2 2.11

github-actions[bot] committed:
Merge staging-next into staging

mweinelt committed:
openldap: 2.6.4 -> 2.6.5

NickCao committed:
Merge pull request #240653 from NickCao/libde265

libde265: 1.0.11 -> 1.0.12

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
Merge pull request #241751 from jtojnar/gnome-staging

GNOME staging updates 2023-07-05

jtojnar committed:
Merge pull request #241939 from jtojnar/libgudev

libgudev: 237 → 238

pacien committed:
mercurial: 6.4.5 -> 6.5

Changelog: https://wiki.mercurial-scm.org/Release6.5

github-actions[bot] committed:
Merge staging-next into staging

Artturin committed:
treewide: go-modules -> goModules

In 787af0f79f128cb21b113d2eb37e64d913f18aae
I had to change ${go-modules} to $goModules to allow overrideAttrs to work;
However, env vars cannot contain -, so  i had to change go-modules too.
This in turn broke nix-update because it uses the go-modules attr.

Instead of making nix-update more complicated, make go-modules naming match cargoDeps.

`fd --type f | xargs sd '\bgo-modules\b' 'goModules'`
and revert change to pkgs/applications/misc/dstask/default.nix
and pkgs/servers/http/dave/default.nix
and pkgs/os-specific/darwin/plistwatch/default.nix

release note added

SuperSandro2000 committed:
Merge pull request #242749 from reckenrode/rustc-darwin-cleanup

rustc: remove Darwin special-casing

zowoq committed:
go_1_20: 1.20.5 -> 1.20.6

Changelog: https://go.dev/doc/devel/release#go1.20

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge staging-next into staging

ajs124 committed:
Merge pull request #242736 from mweinelt/openldap-2.6.5

openldap: 2.6.4 -> 2.6.5

drupol committed:
Merge pull request #242887 from pacien/mercurial-6-5-staging

mercurial: 6.4.5 -> 6.5

drupol committed:
Merge pull request #240939 from PedroHLC/nix-prefetch-git-sri

nix-prefetch-git: add sri-hash to output

ajs124 committed:
libnftnl: 1.2.5 -> 1.2.6

https://lwn.net/Articles/937936/

SuperSandro2000 committed:
libva: 2.18.0 -> 2.19.0

Diff: https://github.com/intel/libva/compare/2.18.0...2.19.0

Changelog: https://raw.githubusercontent.com/intel/libva/2.19.0/NEWS

github-actions[bot] committed:
Merge staging-next into staging

primeos committed:
wayland-protocols: 1.31 -> 1.32

Release announcement:
https://lists.freedesktop.org/archives/wayland-devel/2023-July/042836.html

> This release includes 3 new staging protocols:
>
>   * ext-foreign-toplevel-list - get information about toplevels,
>   * security-context-v1 - allow race free identification of sandboxed
>     clients,
>   * cursor-shape-v1 - set cursor sprite using a shape enum instead of
>     surface.
>
> The xdg-shell protocol also now has a 'suspended' toplevel state,
> usually sent when a toplevel is "out of sight" to the user, meant
> to communicate to a toplevel can for example take power saving measures.
>
> This release also include the usual set of straightening of question
> marks and clarifications of ambiguities.

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
upower: Clean up

- Switch to `finalAttrs` pattern.
- Move `DESTDIR` into `env`.
- Use `lib.optionals` for patches and move the comment inside, in preparation for more patches.

jtojnar committed:
upower: Add installed tests

arianvp committed:
systemd: 253.5 -> 253.6

Allows us to drop a patch

Also fixes systemd-repart being able to run in nix-build

github-actions[bot] committed:
Merge staging-next into staging

PedroHLC committed:
hwdata: 0.371 -> 0.372

github-actions[bot] committed:
Merge staging-next into staging

jtojnar committed:
Merge pull request #243217 from jtojnar/upower

upower: 1.90.0 → 1.90.2

SuperSandro2000 committed:
Merge pull request #242963 from PedroHLC/hwdata-0.372

hwdata: 0.371 -> 0.372

Artturin committed:
Merge pull request #242905 from Artturin/gomodu1

github-actions[bot] committed:
Merge staging-next into staging

Ericson2314 committed:
Merge pull request #243194 from amjoseph-nixpkgs/pr/gcc/crossStageStatic-undo-the-hack

gcc: revert "kludge to prevent mass-rebuild" from #238154

github-actions[bot] committed:
Merge staging-next into staging

vcunat committed:
Merge #242723: libbpf: 1.2.0 -> 1.2.2

...into staging

drupol committed:
Merge pull request #243095 from primeos/wayland-protocols

wayland-protocols: 1.31 -> 1.32

dotlambda committed:
Merge pull request #240480 from dotlambda/pluggy-1.2.0

python310Packages.pluggy: 1.0.0 -> 1.2.0

github-actions[bot] committed:
Merge staging-next into staging

flokli committed:
Merge pull request #243245 from arianvp/bump-systemd

systemd: 253.5 -> 253.6

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #243054 from helsinki-systems/upd/libnftnl

libnftnl: 1.2.5 -> 1.2.6

SuperSandro2000 committed:
Merge pull request #243316 from jpotier/master

ghostscript: 10.01.1 -> 10.01.2

github-actions[bot] committed:
Merge staging-next into staging

SuperSandro2000 committed:
Merge pull request #240604 from tobim/boost-update

boost: 1.79 -> 1.81

github-actions[bot] committed:
Merge staging-next into staging

github-actions[bot] committed:
Merge master into staging-next

blaggacao committed:
rustPlatform.maturinBuildHook: fix postBuild hook to use ./dist contract

github-actions[bot] committed:
Merge master into staging-next

jtojnar committed:
Merge branch 'staging-next' into staging

Conflicts:
 - pkgs/tools/networking/shadowfox/default.nix between e989daa65f2dd9827d56bee6a14a1965471f0597 and 1c29673fcce3aecd0cb4f30058c266ea5480c8a3
 - pkgs/tools/networking/wuzz/default.nix between 7d80417710a9077b9142b479b02d07200d71d9db and 1c29673fcce3aecd0cb4f30058c266ea5480c8a3

github-actions[bot] committed:
Merge master into staging-next

jtojnar committed:
Merge branch 'staging-next' into staging

Conflicts:
- pkgs/servers/geospatial/pg_featureserv/default.nix between 4e68587cdabc75b411ce6e6ada9591f16bf4de20 and 1c29673fcce3aecd0cb4f30058c266ea5480c8a3

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

blaggacao commented on pkgs/build-support/rust/hooks/maturin-build-hook.sh:
```diff
@@ -34,6 +32,9 @@ maturinBuildHook() {
     mkdir -p dist
     mv target/wheels/*.whl dist/
 
+    # These are python build hooks and may depend on ./dist
+    runHook postBuild
+
```
One example: [python-relax-deps-hook](https://github.com/NixOS/nixpkgs/blob/9e17672672a9906a9f313601c6301d4ac7f95580/pkgs/development/interpreters/python/hooks/python-relax-deps-hook.sh#L78), which is a  `postBuild` hook.

SuperSandro2000 committed:
Merge pull request #243059 from SuperSandro2000/libva

libva: 2.18.0 -> 2.19.0

Kranzes committed:
python3Packages.certifi: 2022.12.07 -> 2023.05.07

Diff: https://github.com/certifi/python-certifi/compare/2022.12.07...2023.05.07

SuperSandro2000 committed:
Merge pull request #243811 from Kranzes/certifi

python3Packages.certifi: 2022.12.07 -> 2023.05.07

github-actions[bot] committed:
Merge master into staging-next

github-actions[bot] committed:
Merge staging-next into staging

blaggacao commented:
:hourglass: I'm currently plowing through 1823 rebuilds on my machine and will need to supply (as of now) at least 2 fixes to `maturinBuildHook`-affected packages with (what appears to be) new failures. If you happen to review, please keep this (still) long running local work in mind.
